clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1774,15 +1787,157 @@ private void handleNotControllerError(Errors error) throws ApiException {
+				}
+
+				topicListingFuture.complete(topicListing);
+
+			}
+
+			@Override
+			void handleFailure(Throwable throwable) {
+				topicListingFuture.completeExceptionally(throwable);
+			}
+		}, now);
 
+		Constructor<ListTopicsResult> constructor;
+		ListTopicsResult ltr;
+		try {
+			constructor = ListTopicsResult.class.getDeclaredConstructor(KafkaFuture.class);
+
+			constructor.setAccessible(true);
+			try {
+				ltr = constructor.newInstance(topicListingFuture);
+				return ltr;
+			} catch (InstantiationException | IllegalAccessException | IllegalArgumentException
+					| InvocationTargetException e) {
+				ltr = null;
+				e.printStackTrace();
+			}
+		} catch (NoSuchMethodException | SecurityException e) {
+			e.printStackTrace();
+			constructor = null;
+			ltr = null;
+		}
+		return ltr;
+	}
+	
+	
 	@Override
-	public DescribeTopicsResult describeTopics(final Collection<String> topicNames, DescribeTopicsOptions options) {
-		throw new FeatureNotSupportedException("This feature is not suported for this release.");
+	public DescribeTopicsResult describeTopics(final TopicCollection topics, DescribeTopicsOptions options) {
+
+		if (topics instanceof TopicIdCollection) {
+			throw new FeatureNotSupportedException("This feature is not suported for this release.");
+		}
+
+		final Collection<String> topicNames = ((TopicCollection.TopicNameCollection) topics).topicNames();
+		final Map<String, KafkaFutureImpl<org.apache.kafka.clients.admin.TopicDescription>> topicFutures = new HashMap<>(
+				topicNames.size());
+		final List<String> validTopicNames = new ArrayList<>(topicNames.size());
+
+		for (String topicName : topicNames) {
+			if (topicNameIsUnrepresentable(topicName)) {
+
+				KafkaFutureImpl<org.apache.kafka.clients.admin.TopicDescription> future = new KafkaFutureImpl<>();
+				future.completeExceptionally(new InvalidTopicException(
+						"The given topic name '" + topicName + "' cannot be represented in a request."));
+				topicFutures.put(topicName.toUpperCase(), future);
+
+			} else if (!topicFutures.containsKey(topicName.toUpperCase())) {
+				topicFutures.put(topicName.toUpperCase(),
+						new KafkaFutureImpl<org.apache.kafka.clients.admin.TopicDescription>());
+				validTopicNames.add(topicName.toUpperCase());
+			}
+		}
+		if (validTopicNames.isEmpty()) {
+			org.apache.kafka.clients.admin.DescribeTopicsResult describeTopicsResult = new org.oracle.okafka.clients.admin.DescribeTopicsResult(
+					new HashMap<String, KafkaFuture<org.apache.kafka.clients.admin.TopicDescription>>(topicFutures));
+			return describeTopicsResult;
+		}
+		final long now = time.milliseconds();
+
+		runnable.call(
+				new Call("describeTopics", calcDeadlineMs(now, options.timeoutMs()), new LeastLoadedNodeProvider()) {
+
+					@Override
+					MetadataRequest.Builder createRequest(int timeoutMs) {
+						return new MetadataRequest.Builder(validTopicNames, false, validTopicNames);
```

Comment:
Use DescribeTopic API KEY and DescribeTopicRequest bulder instead of using MetadataRequest builder